crates/worker/src/partition/state_machine/command_interpreter.rs

```diff
@@ -454,6 +473,38 @@ where
         Ok((full_invocation_id, related_span))
     }
 
+    async fn kill_child_invocations<State: StateReader>(
+        &mut self,
+        full_invocation_id: &FullInvocationId,
+        state: &mut State,
+        effects: &mut Effects,
+        journal_length: EntryIndex,
+    ) -> Result<(), Error> {
+        let mut journal_entries = state.get_journal(&full_invocation_id.service_id, journal_length);
+        while let Some(journal_entry) = journal_entries.next().await {
+            let journal_entry = journal_entry?;
+
+            if let JournalEntry::Entry(enriched_entry) = journal_entry {
+                match enriched_entry.header {
+                    // we only need to kill child invocations if they are not completed and the target was resolved
+                    EnrichedEntryHeader::Invoke {
+                        is_completed,
+                        resolution_result: Some(resolution_result),
+                    } if !is_completed => {
+                        let target_fid = FullInvocationId::new(
+                            resolution_result.service_name,
+                            resolution_result.service_key,
+                            resolution_result.invocation_uuid,
+                        );
+                        self.send_message(OutboxMessage::Kill(target_fid), effects);
+                    }
+                    _ => {}
+                }
+            }
```

Comment:
It's still a bit of an open question for me whether we want to kill background calls or not. Perhaps whatever is your opinion, can you write it down in the comment, so next time i stumble on it i'll recall why?

Reply:
At the moment, I am considering background calls and delayed calls somewhat as detached calls which are not part of the call tree. I guess we can add a variant which also allows to kill these calls when killing the parent. I will create a follow-up issue to tackle it and add a comment to the existing code about the current reasoning.

Reply:
https://github.com/restatedev/restate/issues/979